hazelcast/src/main/java/com/hazelcast/wan/impl/InternalWanEvent.java

```diff
@@ -25,8 +25,10 @@
 
 /**
  * Private API for WAN replication events
+ *
+ * @param <T> type of event data
  */
-public interface InternalWanEvent extends WanEvent {
+public interface InternalWanEvent<T> extends WanEvent<T> {
```

Comment:
can we clarify in its javaDoc shortly why we need `InternalWanEvent` and why we can't merge it into `WanEvent`?

Reply:
Fixed.